### ISSUE TITLE
Homekit video / snapshot not loading after a while require reboot of plugin or homebridge

### ISSUE THREAD

**Issue by AMSG22** [hjdhjd/homebridge-unifi-protect#1063]
### Homebridge UniFi Protect Version

V6.20

### Homebridge Platform and OS

Proxmox container LXC 

### Homebridge Version

Ubuntu Lunar homebridge 1.7.0

### Node Version

v20.11.1

### UniFi OS Version

3.2.12

### UniFi Protect Controller Version

3.0.26

### Describe the problem

Hi,
so homekit was working great for the longest time, and maybe in the last few months i noticed the snapshots would not update or take a very long time to update. After rebooting the unifi homebridge plugin it started workin smoothly for a while, haven't really measured how long it takes before it slows down.
so the main change i made was I added a UNVR and use both my UDM SE and UNVR in the protect homebridge controller.
maybe there a setting i have missed or not sure what else i can do.

### Homebridge HBUP JSON configuration

```json
sorry will do this after
```

### Relevant log output

```shell
sorry will do this after.
```

### Acknowledgment that you are only running UniFi OS and UniFi Protect releases from the Ubiquiti Official release channel

- [X] I acknowledge that I am not running any early access / beta firmwares or operating systems from either Ubiquiti or Apple.

**Comment by hjdhjd** [comment]
Ubiquiti has made changes in the way snapshots are generated that make them quite a bit slower, and lower quality. 

I’ve introduced a new high performance snapshot feature that’s enabled by default that address this and produce great results for the vast majority of people, though it requires you to be running on reasonably modern hardware, given it will tax the CPU more. You’re welcome to switch back to the legacy approach to snapshots, utilizing the native Protect APIs, but snapshot performance will suffer and there’s not much I can do there.

TL;DR: you may be running on underpowered hardware trying to do things that require more performance than your system is capable of within the timeframes required by HomeKit. You can optionally revert to the old behavior. Please read the changelog/release notes for HBUP as well as the feature option documentation for more details.

If there’s a **reproducible** bug, please open a new issue with steps to recreate it reliably. You’re also welcome to join the Discord for more interactive support. Thanks!
